processor/processor.go

```diff
@@ -79,6 +79,7 @@ type Handle struct {
 	eventSchemaHandler        types.EventSchemasI
 	dedup                     dedup.Dedup
 	reporting                 types.ReportingI
+	errorReporting            types.ReportingI
```

Comment:
Do we need to expose two separate interfaces in all components? Are cases were we need to decide between `reporting` and `errorReporting`.

On the implementation side, you could still have two different components, but we can develop an abstraction so other parts of the code don't get more complex.

Reply:
> Are cases were we need to decide between reporting and errorReporting

Currently atleast we don't have a case where in we choose between them.

Reply:
In this case, I would suggest the following:

```suggestion
	reporting                 types.ReportingI
```

The rest of the system doesn't need to know about two implementations for reporting. Doing so will make the maintenance and testing for the rest of the code a bit more complex.

Please consider creating a facade that will forward its method calls to both `reporting` and `errorReporting`

Reply:
I have made changes to implement this pattern. Can you once please check this in #3438 ?

Reply:
Sorry closed that one since it had incorrect conflicts. Can you check #3439 ?

Reply:
The PR #3439 has been merged into this one.